### ISSUE TITLE
Incorrect Response Data Information for Terms of Consent Block

### ISSUE THREAD

**Issue by connorhipps** [Automattic/wp-calypso#73347]
### Quick summary

**TL;DR:** The `Terms of Consent Block` in forms pulls the wrong response data information when listed under feedback. The Terms of Consent Block will list a value other than `Consent`. 

This issue will not occur with newly created forms, but the issue does occur with previously created forms in specific scenarios. Please take a look at the ticket for details. 

I've included a video example of this below using a test site. 

https://user-images.githubusercontent.com/5952196/218875707-806ef891-5950-4c91-ae6a-f7d57a76434d.mp4

### Steps to reproduce

1. Start by creating or opening a new site with a Premium Plan. 
2. Select a theme (FSE or otherwise). 
3. Create a new page and open the Code Editor. 
4. Copy and paste the content of any of the form pages noted in this ticket: 5958001-zen
5. Exit the Code Editor, publish/update the page, then preview the page. 
6. Submit the form. 
7. View the site's Feedback section to view a form submission that includes a `Terms Consent block` with the wrong Response Data title. 

### What you expected to happen

It's expected that the `Terms Consent Block` would include Consent beside it. 

### What actually happened

Instead, the `Terms Consent Block` pulls the response data information from a different form field. 

### Impact

Some (< 50%)

### Available workarounds?

Yes

### Platform (Simple and/or Atomic)

Simple

### Logs or notes

- Tested in Brave 1.42.88 and Firefox 109.01
- Theme: Balasana and Blank Canvas

**Comment by connorhipps** [comment]
https://href.li/?https://woothemes.zendesk.com/agent/tickets/5958001

**Comment by connorhipps** [comment]
5958001-zen

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on Simple: Uncertain

📌 FINDINGS/SCREENSHOTS/VIDEO
- By copying the user's form code, I can see the following code for the Consent section of their form:

```
<!-- wp:jetpack/field-consent {"label":"Your Name","required":true,"requiredText":"(required)"} /-->
```

A new form block, meanwhile, shows the following: 

```
<!-- wp:jetpack/field-consent {"requiredText":"(required)"} /-->
```

I reviewed the revision history of the page I pulled the form block from, and they added the Terms Consent block on January 13 2023 - pretty recently.

I have a feeling they duplicated an earlier field, then transformed it into the "Terms Consent" block; that's the way I can 'replicate' the added label on the field.

**Example**

https://user-images.githubusercontent.com/27249804/224521819-d8cde660-667f-469d-bc2d-4f5cd130d08a.mov


📌 ACTIONS
- Since I can't replicate it without taking that step to duplicate a field (which copies over the field's label), I'm thinking this is likely the issue, and there's not a bug happening here. 
- Closing; please reopen if new information points to a possible issue.
